feat: Add doc for node type decorator (#20)
roadiz/docs: src/developer/nodes-system/node_type_decorators.rst

.. _node_type_decorators:

Node-type Decorators
====================

> Roadiz allows you to customize any non-structural properties of node types and their fields.

Reviewer: ```suggestion
Roadiz allows you to customize any node-types and node-type fields **non-structural** properties. This can be handy if you need to change node-types appearence (display names, colors) in Back-office without needing to update your project configuration or emptying caches.
```

Reviewer: Documentation is easier to read if you take examples. Or provide context

Entity representation
---------------------

This decoration is represented by the NodeTypeDecorator entity, which contains three fields :

    - ``path``
    - ``property``
    - ``value``

``path`` property :
^^^^^^^^^^^^^^^^^^^

This property is used to define the *node type* or *node type field* we want to customize.

It consists of the ``node type name`` and the ``node type field name`` separated by a ``dot``.

.. note::
    Exemple of ``path`` for the content field of a Page :

    ``Page.content``

.. warning::
    For decorate an property of a node type simply leave empty after the dot. :

    ``Page.``

``property`` property :
^^^^^^^^^^^^^^^^^^^

This property is used to define the *node type property* or *node type field property* we want to customize.

> It consist of a ``Enum`` who depend if the path contain a field or not.

Reviewer: ```suggestion
It consists of a ``Enum`` which depends on the path containing a field or not.
```

**List of the property for node type :**

#. displayName
#. description
#. color

**List of the property for node type field :**

#. field_label
#. field_universal
#. field_description
#. field_placeholder
#. field_visible
#. field_min_length
#. field_max_length

.. note::

>     Exemple of ``property`` for the content field of a Page :

Reviewer: ```suggestion
    Example of ``property`` for the content field of a Page :
```

    ``field_label``

.. warning::
    You can't attribute a ``node type`` property to a ``path`` with field
    (exemple: path = ``Page.`` and property = ``field_label``)

    You can't attribute a ``node type field`` property to a ``path`` without field
    (exemple: path = ``Page.content`` and property = ``displayName``)

``value`` property :
^^^^^^^^^^^^^^^^^^^

> This property is used to define the value who override default.

Reviewer: ```suggestion
This property is used to override the default value. Default values come from your `node_types/*.yaml` files.
```

`who` is for people, not things

    ``Page.``

``property`` property :
^^^^^^^^^^^^^^^^^^^

This property is used to define the *node type property* or *node type field property* we want to customize.

It consists of a ``Enum`` which depends on the path containing a field or not.

**List of the property for node type :**

#. displayName
#. description
#. color

**List of the property for node type field :**

#. field_label
#. field_universal
#. field_description
#. field_placeholder
#. field_visible
#. field_min_length
#. field_max_length

.. note::
    Example of ``property`` for the content field of a Page :

    ``field_label``

.. warning::
    You can't attribute a ``node type`` property to a ``path`` with field
    (exemple: path = ``Page.`` and property = ``field_label``)

    You can't attribute a ``node type field`` property to a ``path`` without field
    (exemple: path = ``Page.content`` and property = ``displayName``)

``value`` property :
^^^^^^^^^^^^^^^^^^^

This property is used to override the default value. Default values come from your `node_types/*.yaml` files.

It consists of a string linked to its property type.

List of the property type :

>     - **displayName** => text type

Reviewer: ```suggestion
    - **display_name** => text type
```

I suppose this changed after my PR review ?
